3Dmol/glviewer.js

```diff
@@ -1820,6 +1829,10 @@ $3Dmol.GLViewer = (function() {
          */
         this.addSphere = function(spec) {
             spec = spec || {};
+            if(!spec.center.hasOwnProperty("x") || !spec.center.hasOwnProperty("y") || !spec.center.hasOwnProperty("z")){
```

Comment:
I would factor this out as well into the helper function.  And try to make it robust to failure (what if getAtomsFromSel returns nothing? - should provide reasonable defaults).